demos/common/cpp/utils/include/utils/ocv_common.hpp

```diff
@@ -20,43 +20,44 @@
 * @param batchIndex - batch index of an image inside of the blob.
 */
 template <typename T>
-void matU8ToBlob(const cv::Mat& orig_image, const InferenceEngine::Blob::Ptr& blob, int batchIndex = 0) {
+void matToBlob(const cv::Mat& orig_image, const InferenceEngine::Blob::Ptr& blob, int batchIndex = 0) {
     InferenceEngine::SizeVector blobSize = blob->getTensorDesc().getDims();
     const size_t width = blobSize[3];
     const size_t height = blobSize[2];
     const size_t channels = blobSize[1];
     if (static_cast<size_t>(orig_image.channels()) != channels) {
         throw std::runtime_error("The number of channels for net input and image must match");
     }
-    InferenceEngine::LockedMemory<void> blobMapped = InferenceEngine::as<InferenceEngine::MemoryBlob>(blob)->wmap();
-    T* blob_data = blobMapped.as<T*>();
+    // get image value in correct format
+    static const auto img_value = [] (const cv::Mat& img, size_t h, size_t w, size_t c) {
+        switch (img.type()) {
+            case CV_8UC1: return (T)img.at<uchar>(h, w);
+            case CV_8UC3: return (T)img.at<cv::Vec3b>(h, w)[c];
+            case CV_32FC3: return (T)img.at<cv::Vec3f>(h, w)[c];
+        }
+        throw std::runtime_error("Image type is not recognized");
+    };
```

Comment:
Please convert this to separate function. Let's stick to C++ style, not Pythonish.

Reply:
Done